eth/v1/beacon_chain_service.proto

```diff
@@ -0,0 +1,391 @@
+    option (google.api.http) = { get: "/eth/v1/beacon/genesis" };
+  }
+
+  // GetStateRoot calculates HashTreeRoot for state with given 'stateId'. If stateId is root, same value will be returned.
+  rpc GetStateRoot(StateRequest) returns (StateRootResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/root" };
+  }
+
+  // GetStateFork returns Fork object for state with given 'stateId'.
+  rpc GetStateFork(StateRequest) returns (StateForkResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/fork" };
+  }
+
+  // GetFinalityCheckpoints returns finality checkpoints for state with given 'stateId'. In case finality is
+  // not yet achieved, checkpoint should return epoch 0 and ZERO_HASH as root.
+  rpc GetFinalityCheckpoints(StateRequest) returns (StateFinalityCheckpointResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/finality_checkpoints" };
+  }
+
+  // ListValidators returns filterable list of validators with their balance, status and index.
+  rpc ListValidators(StateValidatorsRequest) returns (StateValidatorsResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/validators" };
+  }
+
+  // GetValidator returns a validator specified by state and id or public key along with status and balance.
+  rpc GetValidator(StateValidatorRequest) returns (StateValidatorResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/validators/{validator_id}" };
+  }
+
+  // ListCommittees retrieves the committees for the given state at the given epoch.
+  rpc ListCommittees(StateCommitteesRequest) returns (StateCommitteesResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/states/{state_id}/committees/{epoch}" };
+  }
+
+  // Beacon blocks API related endpoints.
+
+  // GetBlockHeader retrieves block header for given block id.
+  rpc GetBlockHeader(BlockRequest) returns (BlockHeaderResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/headers/{block_id}" };
+  }
+
+  // ListBlockHeaders retrieves block headers matching given query. By default it will fetch current head slot blocks.
+  rpc ListBlockHeaders(BlockHeadersRequest) returns (BlockHeadersResponse) {
+    option (google.api.http) = { get: "/eth/v1/beacon/headers" };
+  }
+
+  // ListBlocks instructs the beacon node to broadcast a newly signed beacon block to the beacon network, to be
+  // included in the beacon chain. The beacon node is not required to validate the signed BeaconBlock, and a successful
+  // response (20X) only indicates that the broadcast has been successful. The beacon node is expected to integrate the
+  // new block into its state, and therefore validate the block internally, however blocks which fail the validation are
+  // still broadcast but a different status code is returned (202).
+  rpc ListBlocks(BeaconBlockContainer) returns (google.protobuf.Empty) {
```

Comment:
Is `ListBlocks` the appropriate name? This doesn't seem correct.

Reply:
Whoops, nice catch.